### ISSUE TITLE
Feature: EarlyExit allow single line nesting

### ISSUE THREAD

**Issue by Levivb** [slevomat/coding-standard#409]
Hey,

After enabling this coding standard on a lot of repositories we (company I work for) noticed about 5% 'silly' early return fixes.

See the following example:
```php
public function __construct(
        PhpEditor $phpEditor,
        JsonEditor $jsonEditor
    ) {
        parent::__construct();

        $this->phpEditor = $phpEditor;
        $this->jsonEditor = $jsonEditor;

        if ($this->phpEditor instanceof CallbackReceiver) {
            $this->phpEditor->setCallback($this->getCallback());
        }
        if (!($this->jsonEditor instanceof CallbackReceiver)) {
            $this->jsonEditor->setCallback($this->getCallback());
        }
    }
```

Which will be fixed to 
```php
public function __construct(
        PhpEditor $phpEditor,
        JsonEditor $jsonEditor
    ) {
        parent::__construct();

        $this->phpEditor = $phpEditor;
        $this->jsonEditor = $jsonEditor;

        if ($this->phpEditor instanceof CallbackReceiver) {
            $this->phpEditor->setCallback($this->getCallback());
        }
        if (!($this->jsonEditor instanceof CallbackReceiver)) {
            return;
        }

        $this->jsonEditor->setCallback($this->getCallback());
    }
```

Although not incorrect, the intent changes from 
```
if this->jsonEditor is a CallbackReceiver: do something
```

to
```
if this->jsonEditor is not a CallbackReceiver: stop with everything
```

The same goes for a simple foreach:
```php
foreach($array as $key => $value){
    if (is_array($value)){
        // do one thing
    }
}
```

which results in 
```php
foreach($array as $key => $value){
    if (!is_array($value)){
        continue;
    }

    // do one thing
}
```

My question here is: do you think it's worth it to add an option for the presence of a minimum amount of nested lines before the early return sniff takes action?

Ofcourse a local `// phpcs:ignore` would work, but I'd like to use those sparingly to avoid codebase cluttering of ignore comments.

Adding an option like `MinimumNestedLines` with a default value of `1` for backwards compatibility would make it possible to reduce the automatic changes of code without removing current functionality.

Let me know what you think. It might be something which only applies to our preferred way of coding, but if you see the benefit it might be nice feature :)

**Comment by mathroc** [comment]
I came here for the same reason, in foreach loop at least I'd like to make early exits optional

**Comment by kukulich** [comment]
@Levivb I'm not able to reproduce the fixer error. See https://github.com/slevomat/coding-standard/commit/2407acb78f7d6934be0d6586aecb131978c8fdf1

About the option... I'm not sure if it's good idea. In both of your examples I still prefere early exit.

**Comment by alcaeus** [comment]
The only point of the sniff is to promote early `return` and `continue` statements over nesting of code paths. For edge cases, you can always disable the sniff on a case-by-case basis using `phpcs:disable`.

**Comment by Levivb** [comment]
@kukulich Ah, there was a typo in the `fixed` example of the constructor.

I didn't mean 
```php
        if (!($this->jsonEditor instanceof CallbackReceiver)) {
            return;
        }

        $this->jsonEditor->setCallback($this->getCallback());
```

but ofcourse without the negation in the if statement:
```php
        if (($this->jsonEditor instanceof CallbackReceiver)) {
            return;
        }

        $this->jsonEditor->setCallback($this->getCallback());
```


This issue/feature was merely pointed towards the allowance of non-early returns if the nested statement only contains a specific number of lines - 1 line in our case.

as @alcaeus suggested, ignoring with `phpcs:disable` could be done. The edge case here, however, is a defined rule of a certain number on lines in the nested statement. So to avoid `phpcs:disable` usage all over the place I thought it would be nice to support it out of the box.

But if I'm the only one benefitting from such a feature it might not be worthwhile to support it. I'll leave that up to you :)
